aten/src/ATen/native/quantized/cpu/conv_serialization.h

```diff
@@ -0,0 +1,248 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <ATen/core/List.h>
+
+#include <ATen/native/quantized/cpu/fbgemm_utils.h>
+#include <ATen/native/quantized/cpu/qnnpack_utils.h>
+
+#include <tuple>
+
+/* Convolution prepacked parameters serialization.
+ *
+ * Files that need to be updated if version changes:
+ * - ATen/native/quantized/cpu/fbgemm_utils.cpp
+ * - jit/passes/onnx/unpack_quantized_weights.cpp
+ *
+ * Version 1
+ *
+ * - Fields:
+ *  1. weight
+ *  2. bias
+ *  3. stride x kSpatialDim
+ *  4. padding x kSpatialDim
+ *  5. dilation x kSpatialDim
+ *  6. groups
+ *
+ * Version 2
+ *
+ * - Fields:
+ *  0. version (string)
+ *  1. list of non-optional tensors
+ *    0: packed parameters (int64_t)
+ *      - kSpatialDim
+ *      - stride x kSpatialDim
+ *      - padding x kSpatialDim
+ *      - dilation x kSpatialDim
+ *      - groups
+ *    1: weight
+ *  2. list of optional tensors
+ *    0: bias
+ *
+ *  Note: version is a string and conv params are packed into a Tensor
+ *    to make ONNX happy (ints and containers of ints are not supported).
+ */
+using ConvParamsSerializationType = std::tuple<
+  // version
+  std::string,
+  // non-optional tensors
+  std::vector<at::Tensor>,
+  // optional tensors
+  std::vector<c10::optional<at::Tensor>>>;
+
+// Parses any historical conv packed params format into
+// the current format.
+template <uint32_t kSpatialDim>
+ConvParamsSerializationType parse_conv_serialized_state(c10::IValue v) {
+
+  // determine the version based on IValue contents
+  int version = -1;
+  if (v.isTuple()) {
+    auto elements = v.toTuple()->elements();
+    if (elements.size() > 0) {
+      auto firstElement = elements[0];
+      if (firstElement.isTensor()) {
+        version = 1;
+      } else if (firstElement.isString()) {
+        std::string version_str = firstElement.toStringRef();
+        // note: not parsing the string to automatically handle bad
+        // inputs
+        if (version_str == "2") {
+          version = 2;
+        }
+      }
+    }
+  }
+  TORCH_INTERNAL_ASSERT(version != -1, "Unable to parse serialization version");
+  TORCH_INTERNAL_ASSERT(version >= 1 && version <= 2, "Unknown serialization version");
```

Comment:
I would skip this since the check above will already fire.  Just one more thing to update if we add another version.